fdbrpc/JsonWebKeySet.cpp

```diff
@@ -0,0 +1,828 @@
+/*
+ * JsonWebKeySet.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "flow/Arena.h"
+#include "flow/AutoCPointer.h"
+#include "flow/Error.h"
+#include "flow/IRandom.h"
+#include "flow/MkCert.h"
+#include "flow/PKey.h"
+#include "flow/UnitTest.h"
+#include "fdbrpc/Base64UrlEncode.h"
+#include "fdbrpc/Base64UrlDecode.h"
+#include "fdbrpc/JsonWebKeySet.h"
+#if defined(HAVE_WOLFSSL)
+#include <wolfssl/options.h>
+#endif
+#include <openssl/bn.h>
+#include <openssl/ec.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <openssl/opensslv.h>
+#include <openssl/rsa.h>
+#include <openssl/x509.h>
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L && !defined(_WIN32)
+#define USE_V3_API 1
```

Comment:
Do we actually need to support different OpenSSL versions?

Reply:
Just thinking forward for when we move beyond v1.x API. 😄
3.x API **deprecates** the API functions for assigning/referencing algo-specific keys to/from EVP_PKEY, so you have to do it with new, algorithm-agnostic API functions.

3.x API has been released in Sep 2021 and is actually part of FDB's Windows CI build. It's just that `EVP_PKEY_get_utf8_string_param()` doesn't work for some reason.